MarkBind/markbind
# Generate .gitignore on init command

**What is the purpose of this pull request?**




**Overview of changes:**

Resolves #1782, and is a breaking change.

TODOs as listed by the original issue:

- [x] Generating a `.gitignore` when `markbind init` is called
- [X] Skip `.gitignore` generation for the `--convert` flag 
- [x] Update UG docs on this behavior
- [x] Add tests as necessary

**Anything you'd like to highlight/discuss:**

This adds a `.gitignore` to the `default` template (only).

The proposed implementation is to skip the `.gitignore` generation when `markbind init` is called with `--convert` flag, but in the future, there may be a need for another action (e.g. merge / overwrite). In this case, please refer to the discussions in #1782 for the rationale on why merging / overwriting may not be a good idea. Also refer to the implementation [here]([4389512](https://github.com/MarkBind/markbind/pull/2248/commits/43895126a74b63543a12fc7894319606c11629a8)) if needed.

The original issue suggests a potential extension to generate a `package.json` as well.

**Testing instructions:**

Pull this PR and call `markbind init`. Check that `.gitignore` is generated when calling `markbind init`, but not when using the `--convert` flag.



**Proposed commit message: (wrap lines at 72 characters)**
```
Generate .gitignore on init command

Currently, when users initialize a site from scratch, we do not
generate .gitignore automatically.

This may cause unnecessary hassle for users using git as they
need to manually create a .gitignore, or add to a pre-existing
.gitignore, all the build files created by markbind.

Generating a .gitignore including common log and build files
from markbind allows users to use our .gitignore without
having to manually find which generated files are unnecessary.

Skipping .gitignore when --convert flag is used prevents
accidentally overwriting a pre-existing .gitignore file.
```


---

**Checklist:** :ballot_box_with_check:
